### ISSUE TITLE
Add support for `foot=use_sidepath` / `sidewalk=separate`

### ISSUE THREAD

**Issue by TheMarex** [Project-OSRM/osrm-backend#4790]
Several communities in the US have started to map sidewalks as separate ways, even if they are attached to the main roads. This degrades the quality of our routing as we route on the main road and the sidewalk, generating erratic routes, problems with snapping and blowing up the size of the road network artificially.

There are two tagging schemes that mark roads that have a detached sidewalk: `foot=use_sidepath` and `sidewalk=separate`. Roads tagged like that should not be included in the routing graph anymore.

**Comment by emiltin** [comment]
ew. mapping sidewalks (and bike lanes) as separate ways is an annoying practice that should be reversed when ever possible :-)

**Comment by TheMarex** [comment]
> mapping sidewalks (and bike lanes) as separate ways is an annoying practice

I agree, but the local communities in Denver, San Jose and Washington DC don't feel that way. This is causing real issues for our pedestrian routing in these cities and ignoring the center-line streets is at least a work-a-round that make it less bad.

**Comment by 1ec5** [comment]
Modeling bike lanes as separate ways is a separate issue from modeling sidewalks as separate ways. The distinction is that sidewalks in these cities generally have “limited access” to the road, whereas bike lanes are part of the road. So in San José, for instance, sidewalks are mapped separately while bike lanes are just `cycleway` tags on the roadways.

**Comment by emiltin** [comment]
not sure what you mean by limited access to the road?

**Comment by 1ec5** [comment]
Sorry for hand-waving there. In much of North America, the norm is that a sidewalk is separated from the roadway by a [verge](https://en.wikipedia.org/wiki/Road_verge), so a router would ideally only have a pedestrian leave the sidewalk and cross the road at crosswalks or perhaps driveways. By contrast, a typical bike lane is only separated from vehicular traffic by a pavement marking. Since the vehicular lanes are modeled together as one way, so would the bike lane be modeled together with them.

As far as I can tell, the presence of a verge is the main reason for modeling sidewalks as separate ways. I’d imagine some [protected bike lane](https://en.wikipedia.org/wiki/Cycle_track) designs would be mapped as a separate way by the same reasoning.

Hopefully instances of `sidewalk=separate` and `foot=use_sidepath` have been mapped carefully enough that avoiding these `highway` ways won’t break routing significantly. I think it’s fair to say the intent of these tags is for walking routers to avoid them anyhow.

**Comment by 1ec5** [comment]
> Perhaps alternatively you can penalize unnamed sidewalks that are along roads? Thus keeps on using the roads?

Penalties could only be applied to graph edges, but the problem that motivated this issue is that the sidewalk ways themselves bloat the routing graph.

> Also sometimes one might want to keep the sidewalk=left and sidewalk=right tags, even though sidewalks are tagged as separate ways. This preserves compatibility with OpenTripPlanner which is moving to support those tags.

How would routers and renderers distinguish this tagging scheme from the scheme in which sidewalks aren’t modeled as separate ways? (Imagine a renderer that synthesizes sidewalks based on `sidewalk=left/right` tags.)

**Comment by 1ec5** [comment]
Should we treat this issue as a duplicate of #1171?

**Comment by nbolten** [comment]
Sidewalks are often mapped as separate footpaths because it's a more appropriate description - they have their own geometries and interface with other features several meters away from the centerline of the road, often with several barriers between the road itself and the sidewalk. Please do not revert them or advocate for reverting them.

I tried to keep my 'reasons to map this way' brief but kept failing, so sorry for the length:

(1) First, geometrical accuracy. Sidewalks are separated ways in some way or another, otherwise they'd just be part of the street - they are separated by markings, bollards, raised curbs, verges, etc. They are several meters removed from the street centerline, and do not need to match the street's geometry with a consistent offset. Whether sidewalk users can also transition to the street varies widely. In addition, as mentioned by @1ec5, they interface with other footways and driveways, and they do so several meters removed from the street centerline.

(2) Second, semantics and connectivity: sidewalks connect to one another via crossings (or other footways), and this is awkward to represent with street centerlines and crossing nodes. Let's take a situation where you want to use crosswalks preferentially. As footways, you can just draw a continuous way and split it into types - sidewalk, crossing, etc, and routing along them is the same approach as for streets, just use the way network + tags. As street metadata, you are in this curious situation: route along the street, keeping in mind what side you are on. When you meet a street, travel 'up' it if it's on the correct 'side' during approach. If you encounter a `highway-crossing` node, use it. Then travel back down the street to the intersection, and turn left/right depending on the initial approach. Continue. There are some ways to attempt to preprocess this information, but at some point this type of logic has to be handled routing with for `sidewalk=*` on street centerlines.

(3) Third, it massively clutters and complicates street centerlines once you have to add even just a little more metadata. Adding just sidewalk width and surface changes means splitting the way every time one of those facts changes, which happens to be pretty often. A semi-randomly selected sidewalk near the University of Washington (my area) has ~3 surface changes and 7 width changes, which would mean splitting the street centerline 10 times. And that's just for one sidewalk - the other side also has a sidewalk that can vary in other important, pedestrian-relevant attributes.

Also, due to reason (2), if a router ever wants to properly handle side-of-street and crossing information, it'll need to expand the graph anyways or embed a lot of extra state into the routing algorithm to keep track of the fake 'turns'.

Finally, there's going to be lots of footways near streets anyways, as OSM becomes more complete, so the challenge of selecting footways vs. streets is there no matter what.

So in summary:

- Separate sidewalks are nifty and great.
- Sidewalks as metadata are fine if your only question is 'is there a sidewalk on this side of the street?', but are a big pain for anything else.
- Tackling this issue is probably necessary even if you disagree with the other points.

**Comment by HolgerJeromin** [comment]
You dont need ugly sidewalk modeling for the need of foot=use_sidewalk
https://www.openstreetmap.org/way/58421500#map=18/51.87161/8.49429
https://www.mapillary.com/app/?focus=photo&pKey=cQvKyObKMGw7MonbaX-X3A

**Comment by jtracey** [comment]
I'll point out that cycle tracks are unambiguously [supposed to be mapped seperately](https://wiki.openstreetmap.org/wiki/Bicycle#Cycle_tracks), and also make use of the similar [bicycle=use_sidepath](https://wiki.openstreetmap.org/wiki/Tag:bicycle%3Duse_sidepath) and [cycleway=separate](https://wiki.openstreetmap.org/wiki/Key:cycleway#Other_values) tags. The former is particularly important, since it means right now OSRM is giving directions that would be illegal to follow.

I'll also add a +1 to mapping sidewalks as separate ways, and note that it's standard practice in Canadian cities. Mapping the interaction between cycleways, sidewalks, and crossings would be a nightmare using tags alone.

**Comment by jdhoek** [comment]
Unlike `bicycle=use_sidepath` — which was [succesfully proposed here](https://wiki.openstreetmap.org/wiki/Proposed_features/use_sidepath), and [is documented here](https://wiki.openstreetmap.org/wiki/Tag:bicycle%3Duse_sidepath) — other tag/value combinations with `use_sidepath` have not been documented and discussed.

Tag usage for `foot=use_sidepath` currently exceeds 30.000 ways.

It would probably help if someone with some affinity with the tags (`foot=use_sidepath` in particular) could document their broader use on any access key (not just `bicycle`) on the relevant wiki-pages, including [access](https://wiki.openstreetmap.org/wiki/Key:access).

To the best of my knowledge `foot=use_sidepath` should mean that OSRM should ignore roads tagged with it for pedestrians and dismounted cyclists — the latter is something the bicycle profile supports, and the use of `foot=use_sidepath` now leads to routing instructions that send cyclists over busy roads they shouldn't use because this tag is ignored, and the road considered suitable for dismounting and walking a bit if it's the faster route.

**Comment by kaligrafy** [comment]
Any news on this?
 I also map sidewalk and cycling paths separately from roads because we need to be more precise in the future around intersections and crossings. Crossing locations and distances can also be used as a routability indicator (minimizing crossing distances also reduces probability of collision with cars).

**Comment by Henryk14** [comment]
@jdhoek @kaligrafy 
Can you look at my pool request?

**Comment by kaligrafy** [comment]
Not stale, still needs discussion and concensus. mapping sidewalks as separate ways will not disappear and is docuemnted officially, so we need to discuss how to deal with foot=use_sidepath. I propose to add it to both the blacklist and the restrited so the routing can use it at the beginning or at the end of the route to connect to a POI or precise location.

**Comment by kaligrafy** [comment]
bump

**Comment by kaligrafy** [comment]
bump

**Comment by Simon0Harms** [comment]
In Germany you aren't allowed to use roads when there are sidewalks.

Therefore, when there is `sidewalk=seperate` and/or `foot=use_sidepath `
[§ 25 StVO](https://www.gesetze-im-internet.de/stvo_2013/__25.html)
> Pedestrians must use the sidewalks. Walking on the road is only permitted if the street has neither a sidewalk nor a shoulder.

(The fine for this is 5€)

For bicycle there is the same rule
[§ 2 IV StVO](https://www.gesetze-im-internet.de/stvo_2013/__2.html)
> There is only an obligation to use cycle paths in the respective direction of travel if this is indicated by signs 237, 240, or 241.

=>  When `bicycle=designated` and/or `bicycle=use_sidepath` is tagged, you aren't allowed to use the street.
(The fine for this is ranged between 20€ and 35€)

=>IMHO at least for Germany it should be changed.

€dit:
And sidewalk:left=separate and sidewalk:right=separate should be considered as well...

**Comment by DennisOSRM** [comment]
Implemented in #7426.
